### PR TITLE
Minimal support for dtype_varying in GPRE for C++

### PR DESCRIPTION
**Pull request by aafemt**
Just in case if someone was going to work with VARBINARY fields from engine in a right way.

### CODE REVIEW

File: src/gpre/cmp.cpp [hunk 1]
```diff
@@ -1731,17 +1731,20 @@ static gpre_port* make_port( gpre_req* request, ref* reference)
 			CPR_bugcheck("missing prototype field for value");
 		if (temp->ref_value && (temp->ref_flags & REF_array_elem))
 			field = field->fld_array;
-		if ((field->fld_length & 7) == 0)
+		FLD_LENGTH l = field->fld_length;
```
Comment (asfernandes) [inline src/gpre/cmp.cpp]: Please don't use variable named just `l`.

**Comment by aafemt** [comment]
Ok. Renamed.

**Comment by AlexPeshkoff** [comment]
I hardly believed my eyes looking at this PR and internals of legacy security DB - they (suppose it was that people who added isc4.gdb to interbase) declared PASSWD field as VARCHAR but always used to work with it as plain CHAR. And I did not notice this bug when used to rework something around for almost 15 years.

First of all - I'm almost sure (did not test) that with your PR legacy authentication will be broken. The problem is that in auth server code field is treated as CHAR - also no VAR at the beginning. Yep, we can fix authenticating code but it will become incompatible with old hashes stored in databases. And that compatibility is the only excuse for existence of kegacy plugin... On my mind we should not raise the hell with changing hash storage format in legacy plugin.

So next question - if we apply the rest of your PR and get support of VAR objects in GPRE will it generate compatible (with old hashes) code for legacy plugin?

**Comment by aafemt** [comment]
> First of all - I'm almost sure (did not test) that with your PR legacy authentication will be broken.

I did test. It worked.

**Comment by AlexPeshkoff** [comment]
Committed PR changes by hand

**Pull request closed without merging**